### ISSUE TITLE
Type safety issue with undefined store value

### ISSUE THREAD

**Issue by itsMapleLeaf** [nanostores/nanostores#116]
In TypeScript, the initial value of an atom is typed as optional. This means the following would typecheck, but fail with a runtime error:
```ts
import { atom } from "nanostores"
import { useStore } from "@nanostores/react"

const message = atom<string>()

export default function App() {
  const currentMessage = useStore(message)
  const firstWord = currentMessage.split(" ")[0]
  return <p>{firstWord}</p>
}
```
[Sandbox link](https://codesandbox.io/s/practical-galileo-zt17m5?file=/src/App.tsx:0-266)

I would suggest either...

- make the initialValue argument required
- add an overload which returns `WritableAtom<T | undefined>`

...although both could be breaking changes for TS users :sweat_smile:

**Comment by ai** [comment]
Yes, good idea. Let’s ask `initialValue` if atom’s type has no `undefined`.

Can I ask you to send PR?

**Comment by itsMapleLeaf** [comment]
Sure! :slightly_smiling_face:

**Comment by ai** [comment]
I decided that explicit way is better 90e7428

**Comment by ai** [comment]
The fix was released in 0.5.13
